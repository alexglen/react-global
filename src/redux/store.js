import { createStore, applyMiddleware, combineReducers, compose } from "redux";
import thunk from "redux-thunk";
import { filterReducer } from "./reducers/filterReducer";
import { modalsReducer } from "./reducers/modalsReducer";
import { moviesReducer } from "./reducers/moviesReducer";
import { searchReducer } from "./reducers/searchReducer";
import { sortingReducer } from "./reducers/sortingReducer";

const reducer = combineReducers({
  moviesReducer,
  filterReducer,
  sortingReducer,
  searchReducer,
  modalsReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(reducer, composeEnhancers(applyMiddleware(thunk)));

export default store;
