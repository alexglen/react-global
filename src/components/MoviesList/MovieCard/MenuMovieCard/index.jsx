import React from "react";
import { useDispatch } from "react-redux";
import { openCardModal, openDeleteModal, setCurrentCardId, setTypeEvent } from "../../../../redux/actions";
import { typeEdit } from "../../../../constants";
import PropTypes from "prop-types";
import "./MenuMovieCard.scss";

function MenuMovieCard({ isMenuModalOpen, setIsMenuModalOpen, movieCardId }) {
  const dispatch = useDispatch();

  const editMovie = () => {
    dispatch(openCardModal());
    dispatch(setCurrentCardId(movieCardId));
    setIsMenuModalOpen(false);
    dispatch(setTypeEvent(typeEdit));
  };

  const deleteMovie = () => {
    dispatch(openDeleteModal());
    dispatch(setCurrentCardId(movieCardId));
    setIsMenuModalOpen(false);
  };

  const close = () => {
    setIsMenuModalOpen(false);
    dispatch(setCurrentCardId(movieCardId));
  };

  const menu = isMenuModalOpen && (
    <div className="menu-modal">
      <ul>
        <li onClick={editMovie}>Edit</li>
        <li onClick={deleteMovie}>Delete</li>
      </ul>
      <div className="close" onClick={close} />
    </div>
  );

  return menu;
}

MenuMovieCard.propTypes = {
  isMenuModalOpen: PropTypes.bool,
  setIsMenuModalOpen: PropTypes.func,
  movieCardId: PropTypes.string,
};

export default MenuMovieCard;
