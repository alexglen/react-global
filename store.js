import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import thunk from 'redux-thunk';
import { filterReducer } from './reducers/filterReducer';
import { modalsReducer } from './reducers/modalsReducer';
import { moviesReducer } from './reducers/moviesReducer';
import { sortingReducer } from './reducers/sortingReducer';

const reducer = combineReducers({
  movies: moviesReducer,
  filters: filterReducer,
  sorting: sortingReducer,
  modals: modalsReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(reducer, composeEnhancers(applyMiddleware(thunk)));

export default store;
