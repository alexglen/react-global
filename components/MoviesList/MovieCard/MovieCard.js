import { useState } from 'react';
import MenuMovieCard from './MenuMovieCard/MenuMovieCard';
import PropTypes from 'prop-types';
import Link from 'next/link';
import styles from './MovieCard.module.scss';

const MovieCard = ({ img, title, releaseDate, genre, id }) => {
  const [isMovieCardWithHover, setIsMovieCardWithHover] = useState(false);
  const [isMenuModalOpen, setIsMenuModalOpen] = useState(false);

  const fullYear = new Date(releaseDate).getFullYear();

  return (
    <article
      className={styles.movie}
      onMouseEnter={() => setIsMovieCardWithHover(true)}
      onMouseLeave={() => setIsMovieCardWithHover(false)}>
      <Link href={`/movies/${encodeURIComponent(id)}`} id='1'>
        <div className={styles.img}>
          <img src={img} alt={title} />
        </div>
      </Link>
      <div className={styles.info}>
        <p>{title}</p>
        <p className={styles.year}>{fullYear}</p>
      </div>
      <p className={styles.genre}>{genre}</p>
      <div
        className={isMovieCardWithHover ? styles.dotsIcon : undefined}
        onClick={() => setIsMenuModalOpen(true)}></div>
      <MenuMovieCard
        isMenuModalOpen={isMenuModalOpen}
        setIsMenuModalOpen={setIsMenuModalOpen}
        movieCardId={id}
      />
    </article>
  );
};

MovieCard.propTypes = {
  img: PropTypes.string,
  title: PropTypes.string,
  releaseDate: PropTypes.string,
  genre: PropTypes.string,
  id: PropTypes.string,
};

export default MovieCard;
