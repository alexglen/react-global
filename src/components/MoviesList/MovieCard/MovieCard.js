import React, { useState } from "react";
import { Link } from "react-router-dom";
import MenuMovieCard from "./MenuMovieCard/MenuMovieCard";
import classnames from "classnames";
import PropTypes from "prop-types";
import "./MovieCard.scss";

const MovieCard = ({ img, title, releaseDate, genre, id }) => {
  const [isMovieCardWithHover, setIsMovieCardWithHover] = useState(false);
  const [isMenuModalOpen, setIsMenuModalOpen] = useState(false);

  const iconClass = classnames("dots-icon", {
    "dots-icon-hidden": !isMovieCardWithHover,
  });

  return (
    <div
      className="movie"
      onMouseEnter={() => setIsMovieCardWithHover(true)}
      onMouseLeave={() => setIsMovieCardWithHover(false)}
    >
      <Link
        to={{
          pathname: `/movies/${title.replace(/\s/g, "")}`,
          id,
        }}
      >
        <div className="movie-img">
          <img src={img} alt={title} />
        </div>
      </Link>
      <div className="movie-info">
        <p>{title}</p>
        <p className="movie-info-year">{new Date(String(releaseDate)).getFullYear()}</p>
      </div>
      <p className="movie-genre">{genre}</p>
      <div className={iconClass} onClick={() => setIsMenuModalOpen(true)}></div>
      <MenuMovieCard isMenuModalOpen={isMenuModalOpen} setIsMenuModalOpen={setIsMenuModalOpen} movieCardId={id} />
    </div>
  );
};

MovieCard.propTypes = {
  img: PropTypes.string,
  title: PropTypes.string,
  releaseDate: PropTypes.string,
  genre: PropTypes.string,
  id: PropTypes.string,
};

export default MovieCard;
