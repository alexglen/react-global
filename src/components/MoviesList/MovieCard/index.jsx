import React, { useState } from "react";
import MenuMovieCard from "./MenuMovieCard";
import { Link } from "react-router-dom";
import classnames from "classnames";
import PropTypes from "prop-types";
import "./MovieCard.scss";

function MovieCard({ img, title, releaseDate, genre, id }) {
  const [isMovieCardWithHover, setIsMovieCardWithHover] = useState(false);
  const [isMenuModalOpen, setIsMenuModalOpen] = useState(false);

  const iconClass = classnames("dots-icon", {
    "dots-icon-hidden": !isMovieCardWithHover,
  });

  const fullYear = new Date(releaseDate).getFullYear();

  return (
    <article
      className="movie"
      onMouseEnter={() => setIsMovieCardWithHover(true)}
      onMouseLeave={() => setIsMovieCardWithHover(false)}
    >
      <Link
        to={{
          pathname: `/movies/${title.replace(/\s/g, "")}`,
          id,
        }}
      >
        <div className="movie-img">
          <img src={img} alt={title} />
        </div>
      </Link>
      <div className="movie-info">
        <p>{title}</p>
        <p className="movie-info-year">{fullYear}</p>
      </div>
      <p className="movie-genre">{genre}</p>
      <div className={iconClass} onClick={() => setIsMenuModalOpen(true)}></div>
      <MenuMovieCard isMenuModalOpen={isMenuModalOpen} setIsMenuModalOpen={setIsMenuModalOpen} movieCardId={id} />
    </article>
  );
}

MovieCard.propTypes = {
  img: PropTypes.string,
  title: PropTypes.string,
  releaseDate: PropTypes.string,
  genre: PropTypes.string,
  id: PropTypes.string,
};

export default MovieCard;
