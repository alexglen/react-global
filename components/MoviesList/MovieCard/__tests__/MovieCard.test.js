import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import store from "../../../../redux/store";
import MovieCard from "../MovieCard";

describe("MovieCard", () => {
  test("render <MovieCard />", () => {
    const component = create(
      <Provider store={store}>
        <BrowserRouter>
          <MovieCard
            img="https://i10.fotocdn.net/s124/a80214fc2d561294/user_xl/2825091898.jpg"
            title="Brother"
            releaseDate="1997-06-13"
            genre="Thriller"
            id="-MWJn2JOvxRpVKrK0Ye3"
          />
        </BrowserRouter>
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
