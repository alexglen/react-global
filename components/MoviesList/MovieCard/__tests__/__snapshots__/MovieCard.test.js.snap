// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard render <MovieCard /> 1`] = `
<article
  className="movie"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    href="/movies/Brother"
    onClick={[Function]}
  >
    <div
      className="movie-img"
    >
      <img
        alt="Brother"
        src="https://i10.fotocdn.net/s124/a80214fc2d561294/user_xl/2825091898.jpg"
      />
    </div>
  </a>
  <div
    className="movie-info"
  >
    <p>
      Brother
    </p>
    <p
      className="movie-info-year"
    >
      1997
    </p>
  </div>
  <p
    className="movie-genre"
  >
    Thriller
  </p>
  <div
    className="dots-icon dots-icon-hidden"
    onClick={[Function]}
  />
</article>
`;
